### PR TITLE
ADBDEV-5273 Use std::nothrow for new allocations

### PR DESCRIPTION
**Pull request by dnskvlnk**
New allocations need to be exception safe, so we replaced regular new operator
by that one that doesn't throw exceptions.

### CODE REVIEW

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/68123

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1233745

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1233746

**Comment by whitehawk** [comment]
Please fix description according to checklist in wiki. For ex, now it is not wrapped to 80 symbols width.

File: gpcontrib/gpcloud/src/gpreader.cpp [hunk 1]
```diff
@@ -33,9 +33,6 @@ unsigned long gpcloud_id_function(void) {
 
 int thread_setup(void) {
     mutex_buf = new pthread_mutex_t[CRYPTO_num_locks()];
```
Comment (whitehawk) [inline gpcontrib/gpcloud/src/gpreader.cpp]: but what if `new` fails to allocate? I think better to change to `new (std::nothrow)`

File: gpcontrib/gpcloud/src/gpreader.cpp [hunk 2]
```diff
@@ -102,9 +99,6 @@ GPReader* reader_init(const char* url_with_options) {
         PrepareS3MemContext(params);
 
         reader = new GPReader(params);
-        if (reader == NULL) {
```
Comment (RekGRpth) [inline gpcontrib/gpcloud/src/gpreader.cpp:105]: https://github.com/arenadata/gpdb/blob/2987c8582ccc878a63b0e69ce2e712e2208810ea/gpcontrib/gpcloud/include/gpreader.h#L50-L51

File: gpcontrib/gpcloud/src/gpwriter.cpp [hunk 3]
```diff
@@ -82,9 +82,6 @@ GPWriter* writer_init(const char* url_with_options, const char* format) {
 
         string extName = params.isAutoCompress() ? string(format) + ".gz" : format;
         writer = new GPWriter(params, extName);
-        if (writer == NULL) {
```
Comment (RekGRpth) [inline gpcontrib/gpcloud/src/gpwriter.cpp:85]: https://github.com/arenadata/gpdb/blob/2987c8582ccc878a63b0e69ce2e712e2208810ea/gpcontrib/gpcloud/include/gpwriter.h#L51-L52

**Comment by RekGRpth** [comment]
update PR description and put title to it

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/68266

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1239270

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1239271

**Review by RekGRpth — APPROVED** [review]
update PR description and put title to it

**Comment by whitehawk** [comment]
Description is still > 80 chars!

**Review by whitehawk — APPROVED** [review]

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/68540

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1252609

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1252611

**Pull request closed without merging**